### DISCUSSION TITLE
Exporting binary formats

### DISCUSSION THREAD

**Discussion by choppsv1** [CESNET/libyang#1564] [Ideas]
I didn't bring this up earlier so that I wouldn't distract from the v2 release. Now that that is basically frozen here's the idea.

For the new ipv[46]_address/prefix types the binary format should be exposed in the API, otherwise users cannot use it directly. Additionally the data creation APIs need to be augmented so that instead of only passing strings I can pass in the binary format.

E.g., if I'm going to be returning the contents of a RIB as some operational data with 1M routes in it I want to be creating things using binary formats, since that's the format I have stored in the RIB which I'm modeling (i.e., struct in_addr or in6_addr). No reason to convert from binary to ascii then back from ascii to binary.

**Comment by michalvasko** [comment]
> For the new ipv[46]_address/prefix types the binary format should be exposed in the API, otherwise users cannot use it directly. 

Right, I was thinking about this too. I mean, you can use the printed LYB value in case it is the same as the stored one (`ipv4/6-address-no-zone`) but it would be better if it can be accessed directly. So fine, I will just withhold this for some short time as it requires a minor version bump.

> Additionally the data creation APIs need to be augmented so that instead of only passing strings I can pass in the binary format.

Already implemented, [this flag](https://github.com/CESNET/libyang/blob/libyang2/src/tree_data.h#L1380) and [this function](https://github.com/CESNET/libyang/blob/libyang2/src/tree_data.h#L1201).
